Fix K8s API server connectivity for IPv6

mKeRix opened the pull request:
The upstream library kube-rs uses an old way of connection to the K8s API, which in their implementation breaks on IPv6 clusters due to wrong URL formatting. This forces the lib to use the proper service, which works on IPv6 clusters as well. The hint towards this solution is from kube-rs/kube-rs#587 - the fix they implemented there is specific to using the rustls feature, which is not in use in this project. But the workaround works regardless.

I also made the env variables configure as a Helm value, in case somebody else needs to change something there in the future. Currently we've applied this fix with kustomize, but having it directly in the chart would let us get rid of that complexity. Plus, others can benefit as well. :)

Commit:
Fix api server connectivity for IPv6

The upstream library kube-rs uses an old way of connection to the K8s API, which in their
implementation breaks on IPv6 clusters due to wrong URL formatting.
This forces the lib to use the proper service, which works on IPv6
clusters as well.

alex-berger commented on charts/rustrial-aws-eks-iam-auth-controller/values.yaml:
```diff
@@ -29,6 +29,12 @@ podSecurityContext:
+env:
+  - name: RUST_LOG
+    value: debug
+  - name: KUBERNETES_SERVICE_HOST
```
I am not sure whether setting `KUBERNETES_SERVICE_HOST` by default is the right thing to do. I am totally with you that we should provide a way (by means of the `env` value) to set it, but we should not set it by default.

What about adding something like this, where it is commented out, but still documented?

```yaml
env:
  # In case your are affected by one of the following bugs or if you are using an IPv6 cluster, 
  # you might need to explicitly set KUBERNETES_SERVICE_HOST to work-around some bugs in kube-rs.
  # - https://github.com/kube-rs/kube-rs/issues/153
  # - https://github.com/kube-rs/kube-rs/issues/587
  #
  #- name: KUBERNETES_SERVICE_HOST
  #  value: kubernetes.default.svc
```

mKeRix replied:
I like the idea & reasoning! The code was updated to move the `RUST_LOG` env variable back and just leave the `env` setting as an optional thing with a note. I chose not to include references to the bugs, as this project isn't affected directly (kube-rs version should be recent enough and rustls is not used). They were rather just a hint to the fix for me when working through their code.

alex-berger commented:
@mKeRix Isn't this fixed in https://github.com/kube-rs/kube-rs/pull/597, if so the proper fix would be to upgrade the latest `kube-rs` version.

alex-berger commented:
> Isn't this fixed in [kube-rs/kube-rs#597](https://github.com/kube-rs/kube-rs/pull/597), if so the proper fix would be to upgrade the latest `kube-rs` version.

_(Note to myself)_ No, it is not fixed in upstream kube-rs , as kube-rs does not properly format IPv6 addresses in URL, it will create something like `https://::1:443` (which is invalid) instead of `https://[::1]:443`.

@mKeRix If you can address [my comment above](https://github.com/rustrial/aws-eks-iam-auth-controller/pull/7#discussion_r787069480), I will happily merge this PR.

Commit:
Make IPv6 fix optional

There's no reason why this would need to be applied to IPv4 cluster, so
instead it can simply be an optional setting with a note.

mKeRix commented:
Indeed. Sorry, I should've made that clearer in my initial post. I have updated the pull request accordingly now. :)

alex-berger committed:
Make log level configurable via values.yaml

alex-berger committed:
Remove hard-coded log level

alex-berger commented:
@mKeRix this PR is now part of today's `v0.1.6` release. Thanks for the PR.
